### PR TITLE
feat!: update API responses to snake_case

### PR DESCRIPTION
**Pull request by HenryQW**
other than camelCase to snake_case for all fields, 

**response** fields changes:
1. totalChipAmount**s** -> total_chip_amount
2. totalChipValue**s** -> total_chip_value
3. nodeAddrs -> node_addresses
4. pnl -> profit_and_loss

**query** changes:
1. item**s**Limit -> item_limit for /epochs/distributions/:transaction_hash
2. staker -> staker_address and node -> node_address for /stake/stakings
3. node_address -> node_addresses for /nodes

**endpoint path** changes:
1. /networks/:networks/list-workers -> /networks/:networks_name/list_workers
2. /nodes/:id/hideTaxRate -> /nodes/:node_address/hide_tax_rate
3. /snapshots/nodes/minTokensToStake ->  /snapshots/nodes/min_tokens_to_stake 
4. /operation/:operator_address/profit -> /nodes/:node_address/operation/profit
5. /snapshots/operators/profit -> /snapshots/nodes/operation/profit
6. /get_endpoint_config -> /endpoint_config

### CODE REVIEW

**Comment by HenryQW** [comment]
Auto-generate OpenAPI with https://github.com/a-h/rest
We can live without examples for now.

**Comment by Songkeys** [comment]
if it's using automated transformation, make sure we don't transform `ID` to `i_d`, `APY` to `a_p_y` etc.

File: internal/database/dialer/cockroachdb/table/staker_profit_snapshot.go [hunk 1]
```diff
@@ -9,14 +9,14 @@ import (
...
-	TotalChipAmounts decimal.Decimal `gorm:"column:total_chip_amounts"`
-	TotalChipValues  decimal.Decimal `gorm:"column:total_chip_values"`
-	CreatedAt        time.Time       `gorm:"column:created_at"`
-	UpdatedAt        time.Time       `gorm:"column:updated_at"`
+	ID              uint64          `gorm:"column:id"`
+	Date            time.Time       `gorm:"column:date"`
+	EpochID         uint64          `gorm:"column:epoch_id"`
+	OwnerAddress    common.Address  `gorm:"column:owner_address"`
+	TotalChipAmount decimal.Decimal `gorm:"column:total_chip_amount"`
+	TotalChipValue  decimal.Decimal `gorm:"column:total_chip_value"`
```
Comment (brucexc) [inline internal/database/dialer/cockroachdb/table/staker_profit_snapshot.go:17]: manually rename the column fields or use a goose migration script to make the changes.

File: internal/service/hub/handler/nta/snapshot_node.go [hunk 2]
```diff
@@ -76,72 +50,93 @@ func (n *NTA) GetStakerProfitSnapshots(c echo.Context) error {
...
-		Limit:      request.Limit,
-		Cursor:     request.Cursor,
-		BeforeDate: request.BeforeDate,
-		AfterDate:  request.AfterDate,
+		Operator: lo.ToPtr(operator),
+		Dates: []time.Time{
+			now.Add(-24 * time.Hour),      // 1 day
+			now.Add(-7 * 24 * time.Hour),  // 1 week
+			now.Add(-30 * 24 * time.Hour), // 1 month
+		},
```
Comment (brucexc) [inline internal/service/hub/handler/nta/snapshot_node.go:109]: why not use https://pkg.go.dev/time#Time.AddDate ?

**Comment by HenryQW** [comment]
> if it's using automated transformation, make sure we don't transform `ID` to `i_d`, `APY` to `a_p_y`, `URL` to `u_r_l` etc.

Yes that won't happen.

I've listed all the breaking changes that may affect the frontend.

**Review by brucexc — APPROVED** [review]

**Review by brucexc — APPROVED** [review]

**Comment by HenryQW** [comment]
pending https://github.com/RSS3-Network/Node/pull/297

**Review by polebug — APPROVED** [review]

**Pull request merged**